workflow/hosts.py

```diff
@@ -47,6 +48,12 @@ def detect(cls):
         elif container is not None:
             machine = 'CONTAINER'
 
+        # TODO: Note that this requires the environment variable to be
+        # defined prior to calling this script; is there a
+        # better/preferred way to do this?
+        elif os.getenv('MACHINE') is not None:
+            machine = os.getenv('MACHINE')
+
```

Comment:
```suggestion
```
Instead of this, how about `machine = os.getenv('MACHINE', "NOTFOUND")` on Line 35 above?